package lightningtracker;

message Address {
	required bool isipv4 = 1 [default = true];

	optional int32 ipv4 = 2;
	optional int64 ipv6a = 3;
	optional int64 ipv6b = 4;

	required int32 port = 5;
}

message HelloToTracker {
	required Address address = 1;
}

message HelloToClient {
	required int32 sessionid = 1;
}

message Ping {
	required int32 pingdata = 1;
}

message Pong {
	required int32 pingdata = 1;
}

message NewBatch {
	enum BatchType {
		RAYTRACE = 0;
		PHOTSHOOTAH = 1;
	}

	required int32 sessionid = 1;
	required string batchhash = 2;
    required int32 tasks = 3;
	required BatchType batchtype = 4;
}

message NodesRequest {
	required int32 sessionid = 1;
	required int32 nodes = 2;
}

message NodesResponse {
	message Node {
		required int32 sessionid  = 1;
		required Address address = 2;
	}
	
	repeated Node nodes = 1;
}

